pkg/gadgettracermanager/containerutils/containerutils.go

```diff
@@ -72,9 +76,33 @@ uint64_t get_cgroupid(char *path) {
 */
 import "C"
 
-type CRIClient interface {
-	Close() error
-	PidFromContainerID(containerID string) (int, error)
+var ContainerRuntimes = []string{
+	docker.Name,
+	containerd.Name,
+	crio.Name,
+}
+
+func NewContainerRuntimeClient(runtime string) (runtimeclient.ContainerRuntimeClient, error) {
+	var runtimeClient runtimeclient.ContainerRuntimeClient
+
+	switch runtime {
+	case docker.Name:
+		runtimeClient = docker.NewDockerClient(docker.DefaultEngineAPISocket)
+	case containerd.Name:
+		runtimeClient = containerd.NewContainerdClient(containerd.DefaultRuntimeEndpoint)
+	case crio.Name:
+		runtimeClient = crio.NewCrioClient(crio.DefaultRuntimeEndpoint)
+	default:
+		return nil, fmt.Errorf("unknown container runtime: %s (available %s)",
+			runtime, strings.Join(ContainerRuntimes, ", "))
+	}
+
+	err := runtimeClient.Initialize()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'm not a big fan of this `Initialize` approach. What about moving this logic to the `NewXXX()` functions? Something like -> https://github.com/kinvolk/inspektor-gadget/commit/54eeee93f553cd3889b9a9476012053af93bbfc2?

Reply:
Yeah, I like your proposal. Applied!